pandas/_libs/interval.pyx

```diff
@@ -414,6 +414,15 @@ cdef class Interval(IntervalMixin):
             return Interval(y.left + self, y.right + self, closed=y.closed)
         return NotImplemented
 
+    def __radd__(self, y):
+        if (
```

Comment:
Well, if you implement infinite recursion, then you get infinite recursion. No surprise here.

Consider extracting a separate *function* for the actual implementation instead, and make both special methods call that, in their own way. Then you can catch the "can't handle that" cases in one place, and otherwise run through the addition code and be done, rather than risking infinite recursion in the first place.